cmd/insprd/operators/channel.go

```diff
@@ -0,0 +1,113 @@
+package operators
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/inspr/inspr/cmd/insprd/memory"
+	"github.com/inspr/inspr/cmd/insprd/memory/brokers"
+	"github.com/inspr/inspr/cmd/insprd/operators/kafka"
+	"github.com/inspr/inspr/cmd/sidecars"
+	"github.com/inspr/inspr/pkg/ierrors"
+	"github.com/inspr/inspr/pkg/meta"
+	metabrokers "github.com/inspr/inspr/pkg/meta/brokers"
+)
+
+//GenOp is a general channel operator for dinamically selecting required operator
+type GenOp struct {
+	brokers brokers.Manager
+	memory  memory.Manager
+	configs map[string]struct {
+		config metabrokers.BrokerConfiguration
+		op     ChannelOperatorInterface
+	}
+}
+
+//NewGeneralOperator creates an instance of GenOp for a given broker and memory manager
+func NewGeneralOperator(brokers brokers.Manager, memory memory.Manager) *GenOp {
+	return &GenOp{
+		brokers: brokers,
+		memory:  memory,
+		configs: make(map[string]struct {
+			config metabrokers.BrokerConfiguration
+			op     ChannelOperatorInterface
+		}),
+	}
+}
+
+func (g GenOp) getOperator(scope string, name string) (ChannelOperatorInterface, error) {
+	channel, _ := g.memory.Channels().Get(scope, name)
+	broker := channel.Spec.SelectedBroker
+
+	config, err := g.brokers.Configs(broker)
+	if err != nil {
+		return nil, err
+	}
+
+	if obj, ok := g.configs[broker]; !reflect.DeepEqual(obj.config, config) || !ok {
+		err = g.setOperator(config)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return g.configs[channel.Spec.SelectedBroker].op, nil
+}
+
+func (g GenOp) setOperator(config metabrokers.BrokerConfiguration) error {
```

Comment:
I think this is unnecessary, as this function is only used in `getOperator` and this validation is already done there